packages/core/data-transfer/src/file/providers/source/index.ts

```diff
@@ -143,13 +149,20 @@ class LocalFileSourceProvider implements ISourceProvider {
               return false;
             }
             return isFilePathInDirname('assets/uploads', filePath);
+            // isFilePathInDirname('assets/metadata', filePath)
           },
-          onentry(entry) {
+          async onentry(entry) {
             const { path: filePath, size = 0 } = entry;
             const normalizedPath = unknownPathToPosix(filePath);
             const file = path.basename(normalizedPath);
-
+            let metadata;
+            try {
+              metadata = await loadAssetMetadata(`assets/metadata/${file}.json`);
+            } catch (error) {
```

Comment:
Is this the handling of old backup files? If yes, we could emit a warning or info diagnostic to warn the user that we're attempting to read assets from an old export & it might have errors or so?

Reply:
Maybe something like "Failed to read metadata for filename, it might be because you're trying to import assets from an old file format or because the export file is corrupted. Strapi will try to fix this issue automatically" (or shorter, or longer, idk)

Reply:
yeah we can do that, it will however always guess that it is a an old format automatically.

Reply:
Damn, I forgot we are in the provider and currently don't have access to the engine diagnostic reporter...

I would avoid printing messages in the console from a provider. The log part should be handled by the implementation based on the diagnostic reports.

I don't think adding this ref between the engine & the provider is part of this PR. Maybe you can comment it & add a todo comment instead?

What do you think?